keyboards/themadnoodle/udon13/keymaps/via/keymap.c

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 The Mad Noodle(@the_mad_noodle)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+/* Layer Names */
+enum layers {
+    _ZERO,
+    _ONE,
+    _TWO,
+    _THREE
+};
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+/*Layer 0*/
+    [_ZERO] = LAYOUT_ortho_3x4(
+  	KC_MUTE, 
+		KC_MPLY, KC_MPRV, KC_MNXT, KC_MSEL, 
+		KC_UNDO, KC_CALC, KC_MAIL, KC_MYCM, 
+		KC_COPY, KC_CUT, KC_PSTE, TO(_THREE)
+        ),
+
+/*Layer 1*/
+    [_ONE] = LAYOUT_ortho_3x4(
+		RGB_TOG, 
+		RGB_SAD, RGB_SAI, RGB_HUD, RGB_HUI, 
+		RGB_MOD, RGB_RMOD, RGB_M_P, RGB_M_B, 
+		RGB_M_SW, RGB_SPD, RGB_SPI, TO(_ZERO)
+        ),
+
+/*Layer 2*/ 
+    [_TWO] = LAYOUT_ortho_3x4(
+		KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, TO(_ZERO)
+        ),
+
+/*Layer 3*/       
+    [_THREE] = LAYOUT_ortho_3x4(
+		KC_NO, 
+		TO(_ONE), TO(_TWO), KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, KC_NO, 
+		KC_NO, KC_NO, KC_NO, TO(_ZERO)
+        ),
+
+};
+
+/*Encoder Mapping*/
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =  {ENCODER_CCW_CW(KC_VOLD,  KC_VOLU)},
+    [1] =  { ENCODER_CCW_CW(RGB_VAD, RGB_VAI)},
+    [2] =  { ENCODER_CCW_CW(KC_LEFT, KC_RGHT)},
+    [3] =  { ENCODER_CCW_CW(KC_DOWN,   KC_UP)},
+   
+};
+#endif
+
+
+
+#ifdef OLED_ENABLE
+
+/* OLED Settings */
+//------------------------------------------------------------------------------------------------
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_0;  // flips the display 180 degrees if offhand
+}
+ 
+
```

Comment:
```suggestion
```

Reply:
Never mind, I was wrong! Sorry about that!